spock-core/src/main/java/org/spockframework/runtime/model/SpecInfo.java

```diff
@@ -168,6 +168,12 @@ public MethodInfo getInitializerMethod() {
     return initializerMethod;
   }
 
+  public Iterable<MethodInfo> getAllInitializerMethods() {
+    if (superSpec == null) return CollectionUtil.listOf(getInitializerMethod());
+
+    return CollectionUtil.concat(superSpec.getAllInitializerMethods(), CollectionUtil.listOf(getInitializerMethod()));
+  }
```

Comment:
I'm not sure if I like this, we create a lot of lists, maybe it makes more sense to have a `collectAll` helper function
```suggestion
  public List<MethodInfo> getAllInitializerMethods() {
    List<MethodInfo> result = new ArrayList<>();
    collectAll(result, SpecInfo::getInitializerMethod);
    return result;
  }
  
  private <T> void collectAll(List<T> result, Function<SpecInfo, T> getter) {
    if (superSpec != null) {
      collectAll(result, getter);
    }
    result.add(getter.apply(this));
  }
```

Reply:
I like that, I just adapted the existing tactic.
But I changed all now to use that accumulation, thx.